app/labor/markdown_fixer.rb

```diff
@@ -47,11 +47,29 @@ def split_tags(markdown)
     end
 
     def underscores_in_usernames(markdown)
-      markdown.gsub(/(@)(_\w+)(_)/, '\1\\\\\2\\\\\3')
+      return markdown unless markdown.include?("@")
```

Comment:
This guard clause has to go through all of `markdown` to determine whether or not to return, so I don't think it saves us much work. If we skip this check we go through the document once, if we keep it we go through the document once + the percentage of text preceding the username.

Reply:
There are number of tests that pass `""`, front matters, or text that do not have username at all (without `@`) to here. These make sense in real world. Hence checking for `@` here to avoid doing work for irrelevant cases and all tests are happy.

Reply:
I understand that the check makes sense, but I think the way it's currently done is slightly inefficient. You already walk the document in the traverser, you could just ignore lines without an `@`, so you only walk the document a total of 1 time. Was that a bit clearer?

Reply:
I understand the reason for the check, my only slight concern was that in articles that contain usernames we actually do slightly more work (first read `markdown` until we find an `@`, then traverse it), but given that the majority of articles will probably not have usernames in them this seems like a good solution.

Reply:
Hey checking against the regexp makes sense. I‘ve updated the code.